### ISSUE TITLE
[BUG/1.20.1] Modded armor ugly render

### ISSUE THREAD

**Issue by Sto3IV** [sp614x/optifine#7605]
## Description of Issue
Armor from different minecraft mods displaying incorrectly: Botania, Ars nouveau, Twilight Forest, Cataclysm.
## Steps to Reproduce
Install Any NeoForge/Forge
Install any of these mods.
Install Optifine.
1. What do you do...
Press E or F5 to look on myself.

## OptiFine Version

OptiFine HD U I6

## Installation Method

Exctracted to mods folder.
## Fabric/Forge Version

NeoForge 17.1.87

## Other Installed Mods
Tested with only Optifine and Twilight Forest, only with ars nouveau+baubles and optifine, also this bug appears with Botania, baubles+optifine.
Now I play with these:
	- additionalstructures 4.2.1
	- aether 1.20.1-1.1.0-neoforge
	- alexsmobs 1.22.7
	- appleskin 2.5.1+mc1.20.1
	- aquaculture 2.5.1
	- aquamirae 6.API15
	- architectury 9.1.12
	- ars_nouveau 4.8.2
	- artifacts 9.2.0
	- ash_api 3.0.2+1.20.1
	- backpacked 2.2.5
	- badpackets 0.4.3
	- bagus_lib 1.20.1-4.2.1
	- balm 7.2.1
	- betteradvancements 0.3.2.161
	- bettercontrols 1.3.0
	- betterdungeons 1.20-Forge-4.0.3
	- bettermineshafts 1.20-Forge-4.0.4
	- betterwitchhuts 1.20-Forge-3.0.3
	- biomesoplenty 18.0.0.598
	- bookshelf 20.1.7
	- boxlib 5.4.2
	- brutalbosses 1.20.1-6.6
	- carryon 2.1.2.7
	- cataclysm 1.0
	- catalogue 1.8.0
	- charmofundying 6.4.5+1.20.1
	- citadel 2.4.9
	- cloth_config 11.1.106
	- collective 7.15
	- configurabledespawntimer 3.2
	- controlling 12.0.2
	- coordinatesdisplay 4.0.0
	- cryingportals 2.4
	- cumulus_menus 1.20.1-1.0.0-neoforge
	- cupboard 1.20.1-2.2
	- curios 5.4.5+1.20.1
	- dash 1.1.1
	- deathfinder 8.0.1
	- defaultoptions 18.0.1
	- dungeons_arise 2.1.57-1.20.1
	- enchdesc 17.0.9
	- endrem 5.2.3-R-1.20.1
	- expandability 9.0.0
	- eyesinthedarkness 1.3.10
	- farlanders 1.4.1
	- forge 47.1.84
	- framework 0.6.16
	- gamemenumodoption 2.2.1
	- gamemenuremovegfarb 2.1.2
	- geckolib 4.2.2
	- gildedarmor 1.7.0-beta.1
	- globalpacks 1.16.2_forge
	- grimoireofgaia 4.0.0-alpha.5
	- hoporp 1.3.7
	- hunters_return 1.20.1-11.2.0
	- infernalmobs 1.20.1.1
	- inventoryhud 3.4.18
	- inventoryprofilesnext 1.10.9
	- ironchest 1.20.1-14.4.4
	- jeed 1.20-2.1.6
	- jei 15.2.0.27
	- jeresources 1.4.0.238
	- justenoughprofessions 3.0.1
	- kotlinforforge 4.8.0
	- lanserverproperties 1.11.1
	- lava_monster 1.0.11
	- libipn 4.0.1
	- mahoutsukai 1.20.1-v1.34.59
	- minecraft 1.20.1
	- mixinextras 0.2.0-beta.9
	- moreoverlays 1.22.1-mc1.20
	- mousetweaks 2.25
	- nitrogen_internals 1.20.1-1.0.1-neoforge
	- nullscape 1.2.4
	- obscure_api 15
	- overflowingbars 8.0.0
	- paintings 0.0NONE
	- patchouli 1.20.1-83-FORGE
	- placebo 8.6.0
	- projectvibrantjourneys 1.20.1-6.0.0
	- puzzlesaccessapi 8.0.7
	- puzzleslib 8.1.11
	- rare_ice 0.0NONE
	- rebind_narrator 2.0.2
	- reblured 1.20.1-1.3.0
	- repurposed_structures 7.1.11+1.20.1-forge
	- searchables 1.0.2
	- skinlayers3d 1.5.4
	- smallernetherportals 3.5
	- spectrelib 0.13.14+1.20.1
	- structory 1.3.4
	- structorytowers 1.0.6
	- structure_gel 2.15.1
	- subterrestrial 2.2.0
	- terrablender 3.0.0.169
	- terralith 2.4.11
	- tlc 1.0.2
	- toastcontrol 8.0.3
	- tramplestopper 5.0.0+mc1.20.1
	- transparent 8.0.1+1.20.1
	- trenzalore 3.3.10
	- tropicraft 9.6.3
	- twilightforest 4.3.1893
	- undergarden 0.8.9
	- waila 8.4.3
	- waveycapes 1.3.2
	- waystones 14.1.1
	- wthit 8.4.3
	- yungsapi 1.20-Forge-4.0.2
	- yungsextras 1.20-Forge-4.0.3 

## Log Files/Crash Reports

https://paste.gg/p/anonymous/3e21235367fc453dbc6bd608b0dffdf4

## F3 Debug Screenshot

https://media.discordapp.net/attachments/650083922029445155/1196831845011177582/2024-01-16_18.01.00.png?ex=65b90f83&is=65a69a83&hm=831f35a7c2e3679e150ac84d67630cb2cc4e99f8b51d57ff48a4006ce4e034e8&=&format=webp&quality=lossless&width=884&height=468

## Additional Information

I reported this bug on some mods GitHub packes, but they told me that if it works without optifine, that it's theproblem with  optifine itself. Because Optifine is closed source so  they can't know exactly what causes the problem.

**Comment by PaleKemono** [comment]
I'm having the same issue

**Comment by Jiingy** [comment]
The screenshot no longer works. Can you provide a new one?

**Comment by Sto3IV** [comment]
![2024-01-16_18 01 00](https://github.com/sp614x/optifine/assets/5441212/0e06b95c-4cdf-4c0c-9004-63e54f221ec2)
Here it is.

**Comment by tvpv2** [comment]
Yes, I play with l'ender cataclysm and ars nouveau mods and I have the same problem. It's not like it can be ignored. Disabling optifine solves the problem, but I am not inclined to disable optifine, which is an important performance and fps mode. You should report this to optifine's development team...

**Comment by tvpv2** [comment]
very understand

**Comment by tvpv2** [comment]
Normal vanilla Minecraft supports both textures where the height and width are equal and where the height is half the width. but optifine may be modifying this. It is annoying that optifine, the most important mod on the market, causes such problems and conflicts with most mods with advanced rendering systems. Fortunately, there is no incompatibility problem with Cracker's wither storm mod. If you are playing this mode, just turn off smooth stack loading in optifine performance settings.
